pkg/build/build.go

```diff
@@ -538,6 +540,26 @@ func WithPackageCacheDir(apkCacheDir string) Option {
 	}
 }
 
+func (b *Build) resolveConfig(ctx context.Context, bc *apko_build.Context) error {
+	ic := bc.ImageConfiguration()
+	pkgs, _, err := bc.BuildPackageList(ctx)
+	if err != nil {
+		return fmt.Errorf("resolving package versions: %w", err)
+	}
+	resolved := make([]string, 0, len(pkgs))
+	for _, pkg := range pkgs {
+		resolved = append(resolved, fmt.Sprintf("%s=%s", pkg.Name, pkg.Version))
+	}
+	ic.Contents.Packages = resolved
+	ic.Archs = []apko_types.Architecture{b.Arch}
+	ic.Contents.Repositories = append(ic.Contents.Repositories, b.ExtraRepos...)
+	ic.Contents.Keyring = append(ic.Contents.Keyring, b.ExtraKeys...)
```

Comment:
This seems fine for now, but can I request that we hoist the resolution / locking logic into a shared place so that this and `tf-apko` and anywhere else we do this don't diverge?

It'd actually also make a great `apko` sub-comment to produce the resolved config (I know we have `show-packages`, but I'm thinking something focused around this logic specifically).